src/layers/sequential.rs

```diff
@@ -59,4 +60,24 @@ where
 
         Tensor::from_shape_vec(output_shape, outputs).unwrap()
     }
+
+    pub fn state_dict(&self) -> BTreeMap<String, Vec<f64>> {
```

Comment:
I believe we discussed how Pytorch returns a `state_dict` where the keys are OrderedDicts right? Maybe instead of having to do this logic and requiring a `weights` and `biases` method in the `Layer` trait it can be combined into one single `state_dict` method per layer, thus the role of Sequential would be to accumulate states across all layers. What do you think?

Reply:
not sure what you mean here. You're suggesting i have a `state_dict `method in the layer trait rather than two methods `weights `and `biases`?

Reply:
Yeah, have you considered it?

Reply:
I see, i'll give that a go. just to make sure we're on the same page. with that approach i'll endup with something like `map<layer_name, state_dict[weight, biases, shape]>`

Reply:
At the sequential layer yes

Reply:
After double checking with @LazaroHurtado, it turns out that PyTorch's state_dict is simply of type `OrderedDict<layer_name, parameters>`, so it doesn't make sense to have `map<layer_name, state_dict[weight, biases, shape]>`. The weights and biases for every layer are stored as separate key-value pairs, just as it is in @Demba3's code. There are no shape keys, but I'm assuming his inclusion of them here is a temporary workaround for this PR.